### PR TITLE
account fixes

### PR DESCRIPTION
**Pull request by tejassonar**
- added filter if available in paginated API calls 
- avoid crash for null donor certificate

### CODE REVIEW

**Comment by sagararyal** [comment]
#### 📝 Changed routes:


| [**/profile/history**](https://planet-webapp-fvadaz47h-planetapp.vercel.app/profile/history) |
|:-:|
|<a href="https://planet-webapp-fvadaz47h-planetapp.vercel.app/profile/history">Screenshot of /profile/history</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/gyNPBPDnve3c64YQBTgbuSPVVU79Hz-W-AqK7GNSVTwlsYfoSxf8YsuZMnogsjuC0WHRLytFzwladop0mNu9Ng.png">(view full size)</a> |







Commit 26e04c07115f3b0f78d38258710dc58d918af7d3 (https://planet-webapp-fvadaz47h-planetapp.vercel.app).

**Review by ankitecd — APPROVED** [review]
Looks good

File: src/features/user/Account/components/AccountRecord.tsx [hunk 1]
```diff
@@ -51,16 +56,16 @@ export default function AccountRecord({
...
+        {(record?.details?.donorCertificate ||
+          record?.details?.taxDeductibleReceipt ||
+          record?.details?.giftCertificate) && (
+          <>
+            <div className={styles.title}>{t('downloads')}</div>
+            <div className={styles.detailGrid}>
+              <Certificates recordDetails={record.details} />
+            </div>
+          </>
+        )}
```
Comment (norbertschuler) [inline src/features/user/Account/components/AccountRecord.tsx:68]: I tested this software with the production backend as I have a conversation donation done there. Unfortunately this of course can not solve the issue if the backend returns a link to a certificate, e.g. https://app.plant-for-the-planet.org/certificate/BYYRKSKQ4891 which then result into an error `TCPDF ERROR: [Image] Unable to get the size of the image: /tmp/certyLJv9q`.

**Comment by norbertschuler** [comment]
Backend filters still do not seem to work correctly (tested on staging and production), e.g. filtering for "failed" does not show any result in the API call besides retrieving all items contains an item with `status: "failed"`:

Bildschirmfoto 2022-04-09 um 11 28 35

I think we should clarify with the backend team once more if these filters are correctly defined:

Bildschirmfoto 2022-04-09 um 11 31 38

**Review by norbertschuler — APPROVED** [review]

**Pull request merged**